src/Settings/Builders/GhcCabal.hs

```diff
@@ -57,21 +57,25 @@ ghcCabalBuilderArgs = mconcat
 -- TODO: Isn't vanilla always built? If yes, some conditions are redundant.
 -- TODO: Need compiler_stage1_CONFIGURE_OPTS += --disable-library-for-ghci?
 -- TODO: should `elem` be `wayUnit`?
+-- This approach still doesn't work. Previously libraries were build only in the 
+-- Default flavours and not using context. 
```

Comment:
This might actually be dead code, since we no longer call `ghc-cabal`. Are you sure this function is actually executed?

Reply:
Oh, maybe this is why this approach is not working. Can you please tell me how exactly we are configuring cabal files then!

Reply:
@chitrak7 I'll try to find time today to remove dead code. You should look into the module `Hadrian.Haskell.Cabal.Parse`.